solr/src/main/scala/akka/stream/alpakka/solr/SolrFlowStage.scala

```diff
@@ -241,10 +245,17 @@ private[solr] final class SolrFlowLogic[T, C](
 
   override def onPush(): Unit = {
     val messagesIn = grab(in)
+    try {
+      sendBulkToSolr(messagesIn)
+      tryPull()
+    } catch {
+      case NonFatal(ex) ⇒
+        decider(ex) match {
+          case Supervision.Stop ⇒ failStage(ex)
+          case _ ⇒ tryPull() // for resume and restart strategies tryPull
```

Comment:
This will drop all messages in `messagesIn`, applying it in `send` could reduce the number of messages dropped.

Reply:
Was that the intended behaviour?

Reply:
This fix allow to use restart or resume strategies  on Supervision.Decider like this:
https://gist.github.com/DmitryBMsk/1ce60ce8f565527f242ca6b4ab70132b
Otherwise any stream may fail during long update process in Solr instead of resuming/restarting.